packages/flame_3d/lib/src/model/model_component.dart

```diff
@@ -0,0 +1,88 @@
+import 'package:flame_3d/camera.dart';
+import 'package:flame_3d/components.dart';
+import 'package:flame_3d/core.dart';
+import 'package:flame_3d/graphics.dart';
+import 'package:flame_3d/src/model/animation_state.dart';
+import 'package:flame_3d/src/model/model.dart';
+
+class ModelComponent extends Object3D {
+  final Model model;
+
+  final Set<int> _hiddenNodes = {};
+  final AnimationState _animation = AnimationState();
+
+  ModelComponent({
+    required this.model,
+    super.position,
+    super.rotation,
+    super.scale,
+  });
+
+  Aabb3 get aabb => _aabb
+    ..setFrom(model.aabb)
+    ..transform(transformMatrix);
+  final Aabb3 _aabb = Aabb3();
+
+  @override
+  void bind(GraphicsDevice device) {
+    final nodes = model.processNodes(_animation);
+    for (final MapEntry(key: idx, value: node) in nodes.entries) {
+      if (_hiddenNodes.contains(idx)) {
+        continue;
+      }
+
+      final mesh = node.node.mesh;
+      if (mesh != null) {
+        device.jointsInfo.jointTransformsPerSurface = node.jointTransforms;
+        // ignore: invalid_use_of_internal_member
```

Comment:
What's happening here?

Reply:
ah yeah that shouldn't be needed anymore